# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/txmgr/TransactionItem.cpp
# Commit: b9ebe58001d7
# Full Hash: b9ebe58001d787bb0bbad6d39ae8681966d5a77f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-04 09:29:13
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r ce2087ddab5c -r b9ebe58001d7 editor/txmgr/TransactionItem.cpp
--- a/editor/txmgr/TransactionItem.cpp	Fri Apr 03 08:30:37 2020 +0000
+++ b/editor/txmgr/TransactionItem.cpp	Fri Apr 03 08:32:01 2020 +0000
@@ -6,6 +6,7 @@
 #include "TransactionItem.h"
 
 #include "mozilla/mozalloc.h"
+#include "mozilla/OwningNonNull.h"
 #include "mozilla/TransactionManager.h"
 #include "mozilla/TransactionStack.h"
 #include "nsCOMPtr.h"
@@ -76,7 +77,8 @@
   if (!mTransaction) {
     return NS_OK;
   }
-  nsresult rv = mTransaction->DoTransaction();
+  OwningNonNull<nsITransaction> transaction = *mTransaction;
+  nsresult rv = transaction->DoTransaction();
   NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
                        "nsITransaction::DoTransaction() failed");
   return rv;
@@ -97,7 +99,8 @@
     return NS_OK;
   }
 
-  rv = mTransaction->UndoTransaction();
+  OwningNonNull<nsITransaction> transaction = *mTransaction;
+  rv = transaction->UndoTransaction();
   if (NS_SUCCEEDED(rv)) {
     return NS_OK;
   }
@@ -271,7 +274,8 @@
     return NS_OK;
   }
 
-  rv = mTransaction->UndoTransaction();
+  OwningNonNull<nsITransaction> transaction = *mTransaction;
+  rv = transaction->UndoTransaction();
   NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
                        "nsITransaction::UndoTransaction() failed");
   return rv;